### PR TITLE
Update public interface and refactor for hooks/function components

### PR DESCRIPTION
**Pull request by scotttrinh**
# Breaking Changes (v1.0.0 candidate!)

1. You must wrap usage of any `FocusGroup` or `KeyHandler` in a `Provider` component. The fallback behavior before was to add handlers to `document.body`, but not only did we not use that feature ourselves, `FocusGroup` doesn't work if you don't have `Provider` present, so it is misleading at best.
2. No more `eventType` prop on `KeyHandler` which only worked for the case where you were not using `Provider`. Since that has gone away, it's not used anywhere anymore. All handlers only fire on the `keydown` event, which was always the case if you had a `Provider`.
3. If multiple `KeyHandler`s share the same trigger in a group context, all matching handlers will fire instead of just the last one to be added. That means if you were previously using the "only-one-handler-can-match" functionality before, you'll probably be firing more events now.
   - For some more context: the idea here is to allow different components that control different parts of the application to respond to the same keydown event even if they're rendered in different parts of the tree. As long as they're on the same "group" level, they'll all fire. This change was actually introduced in #19 
4. Cleaned out a bunch of (IMO) useless module exports, and moved the `KeyHandler` component export to a named export. No more default exports in this package! 🎉 

# Internal changes

The shape of all of the internal changes is mostly to align the style around a more modern function-component style and to use hooks. That had one (surprising) effect on the code: `useEffect` effects fire "bottom-up" so the stack is actually "backwards" now. More of a queue. Since the public interface hasn't changed in this regard, I left the naming the same. We can always rename `this.stack` to `this.queue` if we find that useful, but it seems fine to just look at the head of the "stack" instead of the last element.

One other notable change was to use a React Context to hold the "singleton" of the stack itself rather than using a module side-effect. That is more stable since it avoids side-effects of import order (which forced this change on me), and is more aligned with how you typically see context constructed.

I removed a few spec files that are mostly duplicated by the end-to-end `src/spec.tsx` file.

### CODE REVIEW

File: src/key-handler/index.tsx [hunk 1]
```diff
@@ -41,4 +39,34 @@ export function createMatchingKeyHandler(
   };
 }
 
-export { default } from "./body";
+export function KeyHandler(props: KeyHandlerProps) {
```
Comment (scotttrinh) [inline src/key-handler/index.tsx:42]: This was "moved" from `body.tsx`, but also refactored to a function component, so it's hard to see what actually changed here.

File: src/spec.tsx [hunk 2]
```diff
@@ -1,46 +1,51 @@
...
 describe("<KeyHandler />", () => {
   afterEach(cleanup);
 
-  test("no group: matching key is triggered on the body", () => {
-    const handlerSpy = jest.fn();
-    render(<KeyHandler triggers={[{ key: "KeyA" }]} handler={handlerSpy} />);
-
-    fireEvent.keyDown(document.body, { code: "KeyA" });
-    expect(handlerSpy).toHaveBeenCalledTimes(1);
-  });
```
Comment (scotttrinh) [inline src/spec.tsx:15]: We no longer support the usage of `KeyHandler` without a `Provider`.

File: src/focused-stack/index.ts [hunk 3]
```diff
@@ -52,8 +52,9 @@ class FocusedKeyHandlerStack {
     }
     const key = this.getKeyCodeFromEvent(e);
 
-    const handlerGroup = this.stack[this.stack.length - 1];
+    const handlerGroup = this.stack[0];
```
Comment (mendrew) [inline src/focused-stack/index.ts:55]: Why are we changing this?

As I understand this correctly we have to take the last added group handlers.
Am I wrong?

Comment (mendrew) [reply src/focused-stack/index.ts:55]: Oh, this is because of useEffect fires from the bottom up! Got it!

**Review by mendrew — APPROVED** [review]
Great job! No questions so far, just 👏 !

**Pull request merged**